firehose.c

```diff
@@ -46,11 +46,11 @@
 #include <termios.h>
 #include <time.h>
 #include <unistd.h>
-#include <libxml/parser.h>
-#include <libxml/tree.h>
+
```

Comment:
this is not needed (and does not correspond to what you have in the commit log).

Reply:
libxml/parser.h is unneeded and the libxml/tree.h is included in util.h.
util.h is new header which declares xml parsing wrapper functions, which were previously declared in qdl.h. I don't see by what reason qdl I/O API should be mixed with XML parsing API.

Reply:
I have updated patch with commit message with all reasons.